### PR TITLE
roachtest: use new context to fetch c2c debug zips

### PR DESCRIPTION
**Pull request by msbutler**
In the c2c failures observed in #111952, we were not able to grab the source and destination cluster debug zips because the context used by those calls had already been cancelled, likely by the underlying roachtest infrastructure. This patch changes the fetch debug zip calls to use a new context.

Informs #111952

Release note: none

### CODE REVIEW

File: pkg/cmd/roachtest/tests/cluster_to_cluster.go [hunk 1]
```diff
@@ -588,8 +588,9 @@ func (rd *replicationDriver) setupC2C(
 		rd.t.L().Printf("Prom has started")
 	}
 	return func() {
-		rd.fetchDebugZip(ctx, rd.setup.src.nodes, "source_debug.zip")
-		rd.fetchDebugZip(ctx, rd.setup.dst.nodes, "dest_debug.zip")
+		debugCtx := context.Background()
```
Comment (renatolabs) [inline pkg/cmd/roachtest/tests/cluster_to_cluster.go:591]: Can we set a reasonable timeout on this operation, just to be on the safe side?

**Review by renatolabs — APPROVED** [review]
> likely by the underlying roachtest infrastructure

Yes, that's what happened 👍

File: pkg/cmd/roachtest/tests/cluster_to_cluster.go [hunk 2]
```diff
@@ -588,8 +588,9 @@ func (rd *replicationDriver) setupC2C(
...
+		rd.fetchDebugZip(debugCtx, rd.setup.src.nodes, "source_debug.zip")
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/cluster_to_cluster.go]: Done.

File: pkg/cmd/roachtest/tests/cluster_to_cluster.go (hunk 1 above)
Comment (msbutler) [reply pkg/cmd/roachtest/tests/cluster_to_cluster.go]: Done.

**Comment by msbutler** [comment]
TFTR!

bors r=renatolabs

**Comment by msbutler** [comment]
blathers backport 23.2

**Pull request merged**